server/events_test.go

```diff
@@ -2596,10 +2594,36 @@ func TestServerEventsReload(t *testing.T) {
 	err = json.Unmarshal(msg.Data, &apiResp)
 	require_NoError(t, err)
 
-	require_True(t, apiResp.Data.(string) == "OK")
+	require_True(t, apiResp.Data == nil)
+	require_True(t, apiResp.Error == nil)
 
 	// See that the ping interval has changed.
 	require_True(t, s.getOpts().PingInterval == 200*time.Millisecond)
+
+	// rewrite the config file with a different ping interval
+	err = os.WriteFile(conf, []byte(`garbage and nonsense`), 0666)
+	require_NoError(t, err)
+
+	// Request the server to reload and wait for the response.
+	reply = nc.NewRespInbox()
+	sub, err = nc.SubscribeSync(reply)
+	require_NoError(t, err)
+	err = nc.PublishRequest(subject, reply, nil)
+	require_NoError(t, err)
+	msg, err = sub.NextMsg(time.Second)
+	require_NoError(t, err)
+
+	apiResp = ServerAPIResponse{}
+	err = json.Unmarshal(msg.Data, &apiResp)
+	require_NoError(t, err)
+
+	require_True(t, apiResp.Data == nil)
+	require_True(t, apiResp.Error != nil)
```

Comment:
There is a require_Error() that takes var args for type of error that we expect..

like the following..

```go
require_Error(t, err, errors.New("nats: maximum messages per subject exceeded"))
// or
require_Error(t, err, NewJSReplicasCountCannotBeNegativeError())

```

Reply:
We also have requre_NotNil() IIRC.

Reply:
changed. Any objections to using "github.com/stretchr/testify/require"?

Reply:
Prefer to not introduce that dependency, we have a bunch of our own that work.